src/trio/_path.py

```diff
@@ -32,219 +22,102 @@
-                continue
-            if isinstance(attr, classmethod):
-                wrapper = classmethod_wrapper_factory(cls, attr_name)
-                setattr(cls, attr_name, wrapper)
-            elif isinstance(attr, types.FunctionType):
-                wrapper = thread_wrapper_factory(cls, attr_name)
-                assert isinstance(wrapper, types.FunctionType)
-                wrapper.__signature__ = inspect.signature(attr)
-                setattr(cls, attr_name, wrapper)
-            else:
-                raise TypeError(attr_name, type(attr))
-
-    def generate_magic(cls, attrs: dict[str, object]) -> None:
-        # generate wrappers for magic
-        for attr_name in cls._forward_magic:
-            attr = getattr(cls._forwards, attr_name)
-            wrapper = _forward_magic(cls, attr)
-            setattr(cls, attr_name, wrapper)
-
-    def generate_iter(cls, attrs: dict[str, object]) -> None:
-        # generate wrappers for methods that return iterators
-        wrapper: Callable[..., object]
-        for attr_name, attr in cls._wraps.__dict__.items():
-            if attr_name in cls._wrap_iter:
-                wrapper = iter_wrapper_factory(cls, attr_name)
-                assert isinstance(wrapper, types.FunctionType)
-                wrapper.__signature__ = inspect.signature(attr)
-                setattr(cls, attr_name, wrapper)
-
-
-@final
-class Path(metaclass=AsyncAutoWrapperType):
+class Path(pathlib.PurePath):
     """A :class:`pathlib.Path` wrapper that executes blocking methods in
     :meth:`trio.to_thread.run_sync`.
 
     """
 
-    _forward: ClassVar[list[str]]
-    _wraps: ClassVar[type] = pathlib.Path
-    _forwards: ClassVar[type] = pathlib.PurePath
-    _forward_magic: ClassVar[list[str]] = [
-        "__str__",
-        "__bytes__",
-        "__truediv__",
-        "__rtruediv__",
-        "__eq__",
-        "__lt__",
-        "__le__",
-        "__gt__",
-        "__ge__",
-        "__hash__",
-    ]
-    _wrap_iter: ClassVar[list[str]] = ["glob", "rglob", "iterdir"]
-
-    def __init__(self, *args: StrPath) -> None:
-        self._wrapped = pathlib.Path(*args)
-
-    # type checkers allow accessing any attributes on class instances with `__getattr__`
-    # so we hide it behind a type guard forcing it to rely on the hardcoded attribute
-    # list below.
-    if not TYPE_CHECKING:
-
-        def __getattr__(self, name):
-            if name in self._forward:
-                value = getattr(self._wrapped, name)
-                return rewrap_path(value)
-            raise AttributeError(name)
-
-    def __dir__(self) -> list[str]:
-        return [*super().__dir__(), *self._forward]
+    __slots__ = ()
 
-    def __repr__(self) -> str:
-        return f"trio.Path({str(self)!r})"
+    def __new__(cls: type[PathT], *args: str | os.PathLike[str]) -> PathT:
+        if cls is Path:
+            cls = WindowsPath if os.name == "nt" else PosixPath  # type: ignore[assignment]
```

Comment:
TBH if you just duplicate the return this shouldn't need the type ignore? I think

Reply:
Ah, nope, you just get a different type error.

``` py
def __new__(cls: type[PathT], *args: str | os.PathLike[str]) -> PathT:
        if cls is Path:
            return super().__new__(WindowsPath if os.name == "nt" else PosixPath, *args)
        return super().__new__(cls, *args)
```

error: Incompatible return value type (got "Path", expected "PathT")

And the slightly modified:

```py
def __new__(cls: type[Self], *args: str | os.PathLike[str]) -> Self:
        if cls is Path:
            return super().__new__(WindowsPath if os.name == "nt" else PosixPath, *args)
        return super().__new__(cls, *args)
```

error: Incompatible return value type (got "Path", expected "Self")

I'm not worried about the `type: ignore` here. I'm a big fan of static type checking, but sometimes you've gotta break it in hidden implementations while maintaining the static typing on the public-facing part.